src/terminal/adapter/adaptDispatch.cpp

```diff
@@ -1405,15 +1460,60 @@ bool AdaptDispatch::TabClear(const size_t clearType)
     switch (clearType)
     {
     case DispatchTypes::TabClearType::ClearCurrentColumn:
-        success = _pConApi->PrivateTabClear(false);
+        success = _ClearSingleTabStop();
         break;
     case DispatchTypes::TabClearType::ClearAllColumns:
-        success = _pConApi->PrivateTabClear(true);
+        success = _ClearAllTabStops();
         break;
     }
     return success;
 }
 
+bool AdaptDispatch::_ClearSingleTabStop()
+{
+    CONSOLE_SCREEN_BUFFER_INFOEX csbiex = { 0 };
+    csbiex.cbSize = sizeof(CONSOLE_SCREEN_BUFFER_INFOEX);
+    const bool success = _pConApi->GetConsoleScreenBufferInfoEx(csbiex);
+    if (success)
+    {
+        const auto width = csbiex.dwSize.X;
+        const auto column = csbiex.dwCursorPosition.X;
+
+        _InitTabStopsForWidth(width);
+        _tabStopColumns.at(column) = false;
+    }
+    return success;
+}
+
+bool AdaptDispatch::_ClearAllTabStops() noexcept
+{
+    _tabStopColumns.clear();
+    _initDefaultTabStops = false;
+    return true;
+}
+
+void AdaptDispatch::_ResetTabStops() noexcept
+{
+    _tabStopColumns.clear();
+    _initDefaultTabStops = true;
+}
+
+void AdaptDispatch::_InitTabStopsForWidth(const size_t width)
+{
+    const auto initialWidth = _tabStopColumns.size();
+    if (width > initialWidth)
+    {
+        _tabStopColumns.resize(width);
+        for (auto column = 8u; column < _tabStopColumns.size(); column += 8)
+        {
+            if (_initDefaultTabStops && column >= initialWidth)
```

Comment:
should we move the `if (_initDefaultTabStops` bit earlier, since that's not about to change during the course of this method?

Reply:
Yeah. I was trying to be clever and avoid an extra condition, thinking the compiler would be smart enough to move the check out of the loop. It seems it doesn't do that, though, and it's probably clearer having the condition up front anyway. Will fix this now.

Reply:
Done in commit a0a4780bf0adff3a7829cd2467ce37a9adfd7860.